### PR TITLE
Prepare for configurable dashboards

### PR DESCRIPTION
**Pull request by rsnyman**
- Add model for dashboards
- Add dashboard_id to widget configs
- Create an upgrade system
- Add an initial upgrade to add dashboard_id column
- Add users for future use

### CODE REVIEW

File: backend/ibutsu_server/db/models.py [hunk 1]
```diff
@@ -164,7 +177,66 @@ class WidgetConfig(Model, ModelMixin):
     navigable = Column(Boolean, index=True)
     params = Column(mutable_json_type(dbtype=PortableJSON()))
     project_id = Column(PortableUUID(), ForeignKey("projects.id"), index=True)
```
Comment (john-dupuy) [inline backend/ibutsu_server/db/models.py:179]: This is just the preparation right? This relation will be removed in a later PR?

File: backend/ibutsu_server/db/models.py [hunk 2]
```diff
@@ -164,7 +177,66 @@ class WidgetConfig(Model, ModelMixin):
...
     weight = Column(Integer, index=True)
     widget = Column(Text, index=True)
+
+
+class User(Model, ModelMixin):
+    __tablename__ = "users"
+    email = Column(Text, index=True, nullable=False, unique=True)
+    password = Column(Text, nullable=False)
+    name = Column(Text)
+    project_id = Column(PortableUUID(), ForeignKey("projects.id"), index=True)
```
Comment (john-dupuy) [inline backend/ibutsu_server/db/models.py:192]: I expect a user could be part of many projects. I'm thinking many-to-many here? Or getting rid of the project_id and just having `group_id`?

File: backend/ibutsu_server/db/models.py [hunk 3]
```diff
@@ -164,7 +177,66 @@ class WidgetConfig(Model, ModelMixin):
...
+    group_id = Column(PortableUUID(), ForeignKey("groups.id"), index=True)
+    dashboards = relationship("Dashboard")
+
+
+class Meta(Model):
```
Comment (john-dupuy) [inline backend/ibutsu_server/db/models.py:197]: Could you add a small docblock about this class?

File: backend/ibutsu_server/db/models.py [hunk 4]
```diff
@@ -164,7 +177,66 @@ class WidgetConfig(Model, ModelMixin):
...
+    __tablename__ = "meta"
+    key = Column(Text, primary_key=True, nullable=False, index=True)
+    value = Column(Text)
+
+
+def upgrade_db(session, upgrades):
+    """Upgrade the database using Alembic
+
+    :param session: An SQLAlchemy Session object
+    :param metadata: An SQLAlchemy Metadata object
```
Comment (john-dupuy) [inline backend/ibutsu_server/db/models.py:207]: Can this be removed?

File: backend/ibutsu_server/db/models.py (hunk 1 above)
Comment (rsnyman) [reply backend/ibutsu_server/db/models.py:179]: Yeah, I just left that in there for now. Later on we'll need to do some data migration and move those widgets to a "Default" or some other dashboard.

File: backend/ibutsu_server/db/models.py (hunk 2 above)
Comment (rsnyman) [reply backend/ibutsu_server/db/models.py:192]: Good point, I'll set that up.

File: backend/ibutsu_server/db/models.py [hunk 5]
```diff
@@ -164,7 +177,66 @@ class WidgetConfig(Model, ModelMixin):
...
     title = Column(Text, index=True)
     type = Column(Text, index=True)
     weight = Column(Integer, index=True)
     widget = Column(Text, index=True)
+
+
+class User(Model, ModelMixin):
+    __tablename__ = "users"
+    email = Column(Text, index=True, nullable=False, unique=True)
+    password = Column(Text, nullable=False)
```
Comment (john-dupuy) [inline backend/ibutsu_server/db/models.py:190]: From a cursory google search, thought on using `bcrypt` for storing passwords? 
https://uniwebsidad.com/libros/explore-flask/chapter-12/storing-passwords

File: backend/ibutsu_server/db/models.py (hunk 2 above)
Comment (rsnyman) [reply backend/ibutsu_server/db/models.py]: Converted this to a M-N relationship.

File: backend/ibutsu_server/db/models.py (hunk 5 above)
Comment (rsnyman) [reply backend/ibutsu_server/db/models.py:198]: Yup, we  can. I think I'll leave the actual implementation until we implement auth though.

Comment (john-dupuy) [reply backend/ibutsu_server/db/models.py:198]: Yup, that sounds good to me.

Comment (rsnyman) [reply backend/ibutsu_server/db/models.py]: I did it anyway :-)

**Review by john-dupuy — APPROVED** [review]
LGTM :tada:

**Pull request merged**